source/server/listener_impl.h

```diff
@@ -305,7 +305,10 @@ class ListenerImpl final : public Network::ListenerConfig,
   // Network::ListenerConfig
   Network::FilterChainManager& filterChainManager() override { return filter_chain_manager_; }
   Network::FilterChainFactory& filterChainFactory() override { return *this; }
-  Network::ListenSocketFactory& listenSocketFactory() override { return *socket_factory_; }
+  Network::ListenSocketFactory& listenSocketFactory() override { return *socket_factories_[0]; }
```

Comment:
Will this be eventually removed? Or will it allow returning a specific factory (by index)?

Reply:
Yes, this will be removed until nobody depend on this old interface. I just add a comment here https://github.com/envoyproxy/envoy/pull/21521/files#diff-eb2e967f2ddac82771d5d1948b2f717731c4ff15018d731424b0244cb4639a22R149